components/event-publisher-proxy/pkg/subscribed/helpers.go

```diff
@@ -72,10 +91,49 @@ func AddUniqueEventsToResult(eventsSubSet []Event, uniqEvents map[Event]bool) ma
 }
 
 // FilterEventTypeVersions returns a slice of Events:
+// 1. if the eventType matches the format for standard type matching: <event-name>.<version>
+// E.g. order.created.v0
+// 2. if the eventType matches the format for exact type matching: <eventTypePrefix><appName>.<event-name>.<version>
+// E.g. sap.external.custom.exampleapp.order.created.v0
+func FilterEventTypeVersions(appName string, subscription *eventingv1alpha2.Subscription) []Event {
+	events := make([]Event, 0)
+	for _, eventType := range subscription.Spec.Types {
+		if subscription.Spec.TypeMatching == eventingv1alpha2.TypeMatchingExact {
```

Comment:
I have updated the pseudocode [here](https://github.com/kyma-project/kyma/issues/15624#issuecomment-1342224345) to have one more check. Could you please implement it as well